### ISSUE TITLE
Ido allows exiting early with C-j even when require-match is t

### ISSUE THREAD

**Issue by DarwinAwardWinner** [DarwinAwardWinner/ido-ubiquitous#93]
Try:

```lisp
(ido-completing-read "Prompt: " '("blue" "brown" "yellow" "green") nil t)
```

and then type B and then C-j. `ido-completing-read` will return `"b"` even though that wasn't an element of `collection` and `require-match` was `t`. If possible, ido-cr+ should modify completion behavior to disallow exiting without a match in this way (with the exception of the empty string, which should always be allowed).

ERT test:
```lisp
(should-error 
 (with-simulated-input "b C-j"
   (ido-completing-read "Prompt: " '("blue" "brown" "yellow" "green") nil t)))
```

**Comment by DarwinAwardWinner** [comment]
Fixed by 4306d4ca2a43926f0008c8ec4846f35a09f132cb.

**Comment by DarwinAwardWinner** [comment]
3eccab5 fixes an edge case related to this bug (see the edit in the original description).
